server/liveblog/embed/embed.py

```diff
@@ -28,14 +30,41 @@ def is_relative_to_current_folder(url):
     return not (url.startswith('/') or url.startswith('http://') or url.startswith('https://'))
 
 
+def publish_embed(blog_id, api_host=None, theme=None):
+    html = embed(blog_id, api_host, theme)
+    if not app.config['AMAZON_ACCESS_KEY_ID']:
+        raise AmazonAccessKeyUnknownException()
+    region = app.config['AMAZON_REGION']
+    bucket = app.config['AMAZON_CONTAINER_NAME']
+    s3 = tinys3.Connection(
+        app.config['AMAZON_ACCESS_KEY_ID'],
+        app.config['AMAZON_SECRET_ACCESS_KEY'],
+        default_bucket=bucket,
+        endpoint='s3-%s.amazonaws.com' % (region))
+    # Uploading a single file
+    response = s3.upload('%s/blogs/index.html' % (blog_id), io.BytesIO(bytes(html, 'utf-8')))
+    return response.url.replace('s3-%s.amazonaws.com/%s' % (region, bucket),
+                                '%s.s3-%s.amazonaws.com' % (bucket, region))
+
+
 @bp.route('/embed/<blog_id>')
-def embed(blog_id, api_host=None):
+def embed(blog_id, api_host=None, theme=None):
     api_host = api_host or request.url_root
     blog = get_resource_service('client_blogs').find_one(req=None, _id=blog_id)
     if not blog:
         return 'blog not found', 404
-    theme_root = blog['theme']['name']
+    theme_name = theme
+    try:
```

Comment:
Because `request` can return this error when this method is run outside from a request context.
This is the case when we publish the embed, this celery that calls this method.